vonk/deployment/performance.rst

```diff
@@ -0,0 +1,161 @@
+----------------------------
+
+Because of all the variables mentioned above the best way to find out whether Vonk's performance is sufficient for your use is: Test it yourself.
+
+We provide an evaluation license that you can use for any testing, including performance testing. See :ref:`getting_started`.
+
+Variables
+^^^^^^^^^
+
+Before you start testing, study the variables above and provide answers to them. Then you can configure your platform and your tests in a way that comes closest to the expected real use.
+
+Requests
+^^^^^^^^
+
+You need a set of requests that you want to test. Based on your use case, identify the 5 (or more) most frequent requests. For extra realism you should provide the parameters to the requests from a dataset (like a .csv file with search parameter values).  
+
+What to measure?
+^^^^^^^^^^^^^^^^
+
+There are essentially two questions that you can investigate:
+
+#. Given this deployment, (mix of) requests and an expected request rate, what are the response times?
+#. Given this deployment and (mix of) requests, how many requests can Vonk handle before it starts returning time-outs?
+
+Besides response times more insight can be gained by measuring the load on the server (processor / memory usage, disk and network latency, for both the Vonk Server and the database server).
+
+Based on the answers you can retry with different parameters (e.g. add/remove hardware) to get a sense of the requirements for real use deployment.
+
+Data
+^^^^
+
+Performance testing is best done with data as realistic to your situation as possible. So if you happen to have historic data that you can transform to FHIR Resources, that is the best data to test with.
+
+But if you don't have data of your own, you can use synthesized data. We use data from the Synthea project for our own tests. And we provide :ref:`VonkLoader<vonkloader_index>` to upload the collection bundles from Synthea to Vonk (or any FHIR Server for that matter). 
+
+If you build a Facade, the historical data is probably already in a test environment of the system you build the Facade on. That is a perfect start.
+
+Test framework
+^^^^^^^^^^^^^^
+
+To run performance tests you need a framework to send the requests in parallel and measure the response times. Test automation is a profession in itself so we cannot go into much detail here. You can search for 'REST Performance test tools' to get some options.
+
+Available performance figures
+-----------------------------
+
+We are in the process of setting up performance tests as part of our Continuous Integration and Deployment. Here we describe how this test is currently set up. Because of the beta phase this is in, the output is not yet complete nor fully reliable. Nevertheless we share the preliminary results to give you a first insight.
+
+Vonk performance test setup
+^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Configuration
+
+   #. Repository: MongoDB, both for Administration and for the main database.
```

Comment:
Mongo is used for admin as well?

Reply:
I'm pretty sure it is, will check with Marco.

Reply:
Yep it is, he mentioned it when we were talking about a few days ago.